source/ShopPanel.cpp

```diff
@@ -388,6 +388,15 @@ bool ShopPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 		return SetScrollToTop();
 	else if(key == SDLK_END)
 		return SetScrollToBottom();
+	else if(key == 'k' || (key == 'p' && (mod & KMOD_SHIFT)))
+	{
+		const Ship *flagship = player.Flagship();
+		bool anyEscortUnparked = any_of(playerShips.begin(), playerShips.end(),
+			[&](const Ship *ship){ return ship != flagship && !ship->IsParked() && !ship->IsDisabled(); });
```

Comment:
this should check if the ship is on the same planet/in the same system

Reply:
It iterates over the playerShips list which already excludes those - it only includes ships which have been selected on the right of the shop panels.

That variable is very badly named. It should be something like "selectedPlayerShips".